pages/detail/book/[id]/detail.js

```diff
@@ -13,9 +22,25 @@ export default function DetailBookPage({ data }) {
   if (!currentBook) {
     return <h1>404 An error occured...</h1>;
   }
+
   return (
     <StyledDetail>
-      <Card book={currentBook} />
+      <ModalDelete
+        book={currentBook}
+        onDeleteClick={onDeleteClick}
+        isVisible={isVisible}
+        onToggleVisible={toggleVisible}
+      />
```

Comment:
If you encounter problems with the layering of the modal above all other content, I suggest you to use React.createPortal. This is a method to hook parts of your app not into the #root element on your index.html but into a separate container, for example a #modalWrapper element. By that, you will never have problems displaying the modal above other content. But if it works for now, you can leave it as it is.

Reply:
Maybe I'll change it again later.